### PR TITLE
Update DG user stories, use cases, glossary and NFR Amanda

### LINKED ISSUE #277: Finalise DG Amanda

### PR DESCRIPTION
**Pull request by github-amanda**
1. Modify existing user stories and existing use cases by adding extensions and editing MSS
3. Add user stories and use cases for new functionalities that were not addressed
3. Add non-functional requirements and some terms in glossary for more clarity

Close #277

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -645,17 +645,17 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 | `* * *` | user who stocks up items regularly | update my items’ details | change the items' expiry dates and/or quantities accordingly | 
-| `* * *` | user with many items | list down all items | know all the items I have in one glance |
-| `* *` | user who has many items in my room | see all items in my room | keep track of exactly what I have | 
-| `* *` | user who tags my items meaningfully | see all items with the same tag | see what items I have with that particular tag |
-| `* * *` | forgetful user with many items | search for an item quickly | locate them easily |
+| `* * *` | user with many items | list down all my items | know all the items I have in one glance |
+| `* * *` | forgetful user with many items | find an item quickly using its name | locate them easily |
+| `* *` | user who has many items in the storeroom | view all items in the storeroom | keep track of exactly what I have | 
+| `* *` | user who keeps track of my guests' favourites | view all items tagged with "guests" | better prepare the items when hosting my guests |
 | `* *` | forgetful user | be aware of my expiring items | replace them before it is expired or discard them if they have expired |
```
Comment (kumsssss) [inline docs/DeveloperGuide.md]: I feel like this overlaps with line 656. Maybe can consider removing it?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -756,19 +739,19 @@ otherwise)
...
-2. StoreMando displays a list of items that have either expired or are expiring within the next 7 days.
+1. User wants to search for items that are expiring within a specific number of days.
+2. StoreMando displays a list of items that have either expired or are expiring within the specified number of days.
 
    Use case ends.
 
 **Extensions**
 
-* 1a. User inputs a negative number.
+* 1a. User inputs a number smaller than -365 or larger than 365.
```
Comment (kumsssss) [inline docs/DeveloperGuide.md:765]: I can't comment above but perhaps you should do this for add and edit too for the quantity limit if you are doing it for reminder.

**Review by kumsssss — COMMENTED** [review]
Minor changes to fix. Otherwise, LGTM!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -842,8 +851,6 @@ Use case ends.
 
       Use case resumes at step 1.
 
-*{More to be added}*
-
 ### Non-Functional Requirements
```
Comment (Md-Fazil) [inline docs/DeveloperGuide.md:847]: If you have not, I'd suggest to review this section as well. If you have done this and found that no changes are needed, ignore this comment.

**Review by Md-Fazil — APPROVED** [review]
LGTM!

**Pull request merged**